driver-core/src/main/com/mongodb/connection/InternalStreamConnection.java

```diff
@@ -200,9 +198,7 @@ public void sendMessage(final List<ByteBuf> byteBuffers, final int lastRequestId
 
         writerLock.lock();
         try {
-            int messageSize = getMessageSize(byteBuffers);
             stream.write(byteBuffers);
-            connectionListener.messagesSent(new ConnectionMessagesSentEvent(getId(), lastRequestId, messageSize));
```

Comment:
Technically, a compatibility break for anyone using com.mongodb.connection.ClusterFactory#create directly and passing a non-null ConnectionListener.  But I don't think anyone will notice.